src/main/java/org/jboss/xavier/integrations/rbac/RBACRouteBuilder.java

```diff
@@ -0,0 +1,149 @@
+
+                .process(exchange -> {
+                    // save decoded x-rh-identity JsonNode as header
+                    String xRHIdentity = exchange.getIn().getHeader(RBAC_X_RH_IDENTITY, String.class);
+                    JsonNode xRHIdentityDecodedJsonNode;
+                    try {
+                        xRHIdentityDecodedJsonNode = new ObjectMapper().reader().readTree(
+                                new String(
+                                        Base64.getDecoder().decode(xRHIdentity)
+                                )
+                        );
+                    } catch (IOException e) {
+                        Logger.getLogger(this.getClass().getName()).warning("Unable to read " + RBAC_X_RH_IDENTITY);
+                        throw new IllegalStateException("Could not read header " + RBAC_X_RH_IDENTITY);
+                    }
+                    exchange.getIn().setHeader(RBAC_X_RH_IDENTITY_DECODED, xRHIdentityDecodedJsonNode);
+                })
+                .process(exchange -> {
+                    // isOrgAdmin
+                    JsonNode xRHIdentity = exchange.getIn().getHeader(RBAC_X_RH_IDENTITY_DECODED, JsonNode.class);
+                    JsonNode isOrgAdmin = xRHIdentity.get("identity").get("user").get("is_org_admin");
+                    exchange.getIn().setHeader(RBAC_IS_ORG_ADMIN, isOrgAdmin.booleanValue());
+                })
+                .choice()
+                    .when(exchange -> exchange.getIn().getHeader(RBAC_IS_ORG_ADMIN, Boolean.class))
+                        .setHeader(RBAC_USER_ACCESS, constant(null))
+                    .endChoice()
+                    .otherwise()
+                        .enrich("direct:fetch-rbac-user-access", (oldExchange, newExchange) -> {
+                            List<Acl> acls = newExchange.getIn().getBody(List.class);
+                            Map<String, Map<String, List<String>>> accessForUser = RBACService.getAccessForUser(acls);
+
+                            oldExchange.getIn().setHeader(RBAC_USER_ACCESS, accessForUser);
+                            return oldExchange;
+                        })
+                    .endChoice()
+                .end();
+
+        from("direct:fetch-rbac-user-access")
+                .routeId("fetch-rbac-user-access")
+                .setHeader("access", ArrayList::new)
+                .setHeader("nextLink", () -> "")
+                .loopDoWhile(exchange -> exchange.getIn().getHeader("nextLink") != null)
+                    .setHeader(Exchange.HTTP_METHOD, constant(HttpMethods.GET))
+                    .setHeader(Exchange.CONTENT_TYPE, constant("application/json"))
+                    .setHeader(Exchange.HTTP_PATH, constant(rbacPath))
+                    .process(exchange -> {
+                        String httpQuery;
+                        String nextLink = exchange.getIn().getHeader("nextLink", String.class);
+                        int queryParamsIndex = nextLink.indexOf("?");
```

Comment:
For just one character there's the "dedicated" `indexOf(int ch)` method
```suggestion
                        int queryParamsIndex = nextLink.indexOf('?');
```

Reply:
done